python/ray/data/datasource/bigquery_datasink.py

```diff
@@ -121,5 +122,5 @@ def _write_single_block(block: Block, project_id: str, dataset: str) -> None:
 
         # Launch a remote task for each block within this write task
         for block in blocks:
-            _write_single_block.remote(block, self.project_id, self.dataset)
+            ray.get(_write_single_block.remote(block, self.project_id, self.dataset))
```

Comment:
We'll need to call `ray.get` on all of the refs at the same time.
```python
ray.get([_write_single_block.remote(block, self.project_id, self.dataset) for block in blocks])
```

Otherwise, the tasks run serially.

Reply:
good catch, fixed now!